phpseclib/Crypt/RSA.php

```diff
@@ -2053,17 +2033,20 @@ function _rsassa_pkcs1_v1_5_verify($m, $s)
         $s = $this->_os2ip($s);
         $m2 = $this->_rsavp1($s);
         if ($m2 === false) {
-            throw new \RuntimeException('Invalid signature');
+            return false;
         }
         $em = $this->_i2osp($m2, $this->k);
         if ($em === false) {
-            throw new \RuntimeException('Invalid signature');
+            return false;
         }
 
         // EMSA-PKCS1-v1_5 encoding
 
-        $em2 = $this->_emsa_pkcs1_v1_5_encode($m, $this->k);
-        if ($em2 === false) {
+        // If the encoding operation outputs "intended encoded message length too short," output "RSA modulus
+        // too short" and stop.
+        try {
+            $em2 = $this->_emsa_pkcs1_v1_5_encode($m, $this->k);
+        } catch (\LengthException $e) {
```

Comment:
For all practical purposes `_emsa_pkcs1_v1_5_encode()` can't return false. http://php.net/manual/en/function.hash.php doesn't return false in the master branch (where Hash.php was rewritten to be a wrapper for PHP's builtin hash extension) and even when phpseclib had it's own implementations of select hash algorithms none of them appear to return false either:

https://github.com/phpseclib/phpseclib/blob/1.0/phpseclib/Crypt/Hash.php

Here's what the RFC says:

```php
   1. Apply the hash function to the message M to produce a hash value
      H:

         H = Hash(M).

      If the hash function outputs "message too long," output "message
      too long" and stop.
```
According to https://en.wikipedia.org/wiki/Template:Comparison_of_SHA_functions you'd have to have a message 2048 petabytes long for that error to be output and if you're dealing with data of that length PHP is probably not the right language for you lol.

In light of this I've just removed the whole `return false` thing.